templating.rst

```diff
@@ -6,74 +6,77 @@ Creating and Using Templates
-CSS or any other content, it hands the work off to the templating engine.
+the controller delegates most of the heavy work to other places. When
+a controller needs to generate HTML, it hands the work off to the
+*templating* engine.
+
 In this chapter, you'll learn how to write powerful templates that can be
 used to return content to the user, populate email bodies, and more. You'll
 learn shortcuts, clever ways to extend templates and how to reuse template
 code.
 
+Rendering a Template
+--------------------
+
+Typically, you'll render a template from inside if your controller::
+
+    // src/AppBundle/Controller/LuckyController.php
+
+    public function numberAction()
+    {
+        $number = mt_rand(0, 10);
+
+        // renders app/Resources/views/lucky/number.html.twig
+        return $this->render('lucky/number.html.twig', array(
+            // pass a variable called randomNumber into the template
+            'randomNumber' => $number,
+            'people' => array('Javier', 'Christian, 'Wouter')
+        ))
+    }
+
+This template passes two variables - ``randomNumber`` and ``people`` into your
+template. A very simple template, might look like this:
+
+.. code-block:: html+twig
+
+    {# app/Resources/views/lucky/number.html.twig #}
+
+    <h1>Your lucky number: {{ randomNumber }}</h1>
+
+    <h3>Nice People:</h3>
+    <ul>
+        {% for person in people %}
+            <li>
+                {{ person }}
+            </li>
+        {% endfor %}
+    </ul>
+
+Of course, this is only the beginning! Keep reading to learn more about
+`Twig`_, the templating engine used by Symfony!
+
 .. note::
 
-    How to render templates is covered in the
+    For more details on rendering templates, see the
     :ref:`controller <controller-rendering-templates>` article.
 
-.. index::
-   single: Templating; What is a template?
-
-Templates
----------
-
-A template is simply a text file that can generate any text-based format
-(HTML, XML, CSV, LaTeX ...). The most familiar type of template is a *PHP*
-template - a text file parsed by PHP that contains a mix of text and PHP code:
-
-.. code-block:: html+php
-
-    <!DOCTYPE html>
-    <html>
-        <head>
-            <title>Welcome to Symfony!</title>
-        </head>
-        <body>
-            <h1><?php echo $page_title ?></h1>
-
-            <ul id="navigation">
-                <?php foreach ($navigation as $item): ?>
-                    <li>
-                        <a href="<?php echo $item->getHref() ?>">
-                            <?php echo $item->getCaption() ?>
-                        </a>
-                    </li>
-                <?php endforeach ?>
-            </ul>
-        </body>
-    </html>
-
-.. index:: Twig; Introduction
-
-But Symfony packages an even more powerful templating language called `Twig`_.
-Twig allows you to write concise, readable templates that are more friendly
-to web designers and, in several ways, more powerful than PHP templates:
```

Comment:
But the documentation is about the component right? The templating logic from the full stack is going to be deprecated, so is it worth it? Maybe would it be better to document the `TwigBundle` and the `TwigBridge` instead?

With the new navigation its not that easy to find some documentation about twig and doctrine integration. Only the `PHPUnit Bridge` is [among the components](http://symfony.com/doc/current/components/index.html), and there is none the bundles from the symfony repo [among the list of bundles](http://symfony.com/doc/bundles/). Documenting them would be a also a place for some new and old cookbooks to link there.

Maybe this is worth a new issue, wdyt?